process/fuser.rs

```diff
@@ -0,0 +1,1107 @@
+    let reader = io::BufReader::new(file);
+
+    for line in reader.lines() {
+        let line = line?;
+        let parts: Vec<&str> = line.split_whitespace().collect();
+
+        if let (Some(net_inode_str), Some(scanned_path)) = (parts.get(6), parts.get(7)) {
+            let net_inode = net_inode_str.parse().unwrap_or(0);
+            let path = normalize_path(scanned_path);
+
+            match timeout(&path, 5) {
+                Ok(stat) => UnixSocketList::add_socket(
+                    unix_socket_list,
+                    scanned_path.to_string(),
+                    stat.st_dev,
+                    stat.st_ino,
+                    net_inode,
+                ),
+                Err(_) => continue,
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Reads the `/proc/mounts` file and populates the provided `MountList` with mount points.
+fn read_proc_mounts(mount_list: &mut MountList) -> io::Result<&mut MountList> {
+    let file = File::open(PROC_MOUNTS)?;
+    let reader = io::BufReader::new(file);
+
+    for line in reader.lines() {
+        let line = line?;
+        let parts: Vec<&str> = line.split_whitespace().collect();
+        let mountpoint = PathBuf::from(parts[1].trim());
+        mount_list.mountpoints.push(mountpoint);
+    }
+
+    Ok(mount_list)
+}
+
+/// Normalizes a file path by removing the leading '@' character if present.
+fn normalize_path(scanned_path: &str) -> String {
+    if scanned_path.starts_with('@') {
+        scanned_path[1..].to_string()
+    } else {
+        scanned_path.to_string()
+    }
+}
+
+/// Parses network socket information from the `filename` field of the `Names` struct
+/// and returns an `IpConnections` instance.
+fn parse_inet(names: &mut Names, ip_list: &mut IpConnections) -> io::Result<IpConnections> {
+    let filename_str = names.filename.to_string_lossy();
+    let parts: Vec<&str> = filename_str.split('/').collect();
+
+    if parts.len() < 2 {
+        return Err(Error::new(
+            ErrorKind::InvalidInput,
+            "Invalid filename format",
+        ));
+    }
+
+    let protocol = parts[1];
+    let hostspec = parts[0];
+    let host_parts: Vec<&str> = hostspec.split(',').collect();
+
+    let lcl_port_str = host_parts
+        .get(0)
+        .ok_or_else(|| Error::new(ErrorKind::InvalidInput, "Local port is missing"))?;
+    let rmt_addr_str = host_parts.get(1).cloned();
+    let rmt_port_str = host_parts.get(2).cloned();
+
+    let lcl_port = lcl_port_str
+        .parse::<u64>()
+        .map_err(|_| Error::new(ErrorKind::InvalidInput, "Invalid local port format"))?;
+
+    let protocol_socktype = match protocol {
+        "tcp" => libc::SOCK_STREAM,
+        "udp" => libc::SOCK_DGRAM,
+        _ => {
+            return Err(Error::new(ErrorKind::InvalidInput, "Unsupported protocol"));
+        }
+    };
+
+    if rmt_addr_str.is_none() && rmt_port_str.is_none() {
+        let rmt_addr = IpAddr::V4("0.0.0.0".parse().unwrap());
```

Comment:
Please remove unwrap(), and this also applies to all places where it is present and not appropriate.